test/export-api-def.test.js

```diff
@@ -0,0 +1,116 @@
+/*global describe, beforeEach, it */
+'use strict';
+var path = require('path');
+var helpers = require('yeoman-generator').test;
+var fs = require('fs');
+var expect = require('chai').expect;
+var wsModels = require('loopback-workspace').models;
+var common = require('./common');
+var yaml = require('js-yaml');
+var install = require('strong-cached-install');
+var SANDBOX =  path.resolve(__dirname, 'sandbox');
+var PKG_CACHE = path.resolve(__dirname, '..', '.pkgcache');
+
+describe('loopback:export-api-def generator', function() {
+  this.timeout(30 * 60 * 1000); // 30 minutes
+  before(common.resetWorkspace);
+
+  before(function createSandbox(done) {
+    helpers.testDirectory(SANDBOX, done);
+  });
+
+  before(function createProject(done) {
+    common.createDummyProject(SANDBOX, 'test-app', done);
+  });
+
+  before(function createProductModel(done) {
+    var test = this;
+    wsModels.ModelDefinition.create(
+      {
+        name: 'Product',
+        facetName: 'common'
+      },
+      function(err, model) {
+        test.Model = model;
+        done(err);
+      });
+  });
+
+  before(function installDependencies(done) {
+    install(SANDBOX, PKG_CACHE, ['dependencies'], done);
+  });
+
+  it('JSON format', function(done) {
+    var gen = givenGenerator();
+    gen.options.output = 'swagger.json';
+
+    gen.run(function() {
+      var content = readJsonFile();
+      expect(content).to.have.property('swagger', '2.0');
+      expect(content).to.not.have.property('public');
+      expect(content).to.have.property('basePath');
+      expect(content).to.have.property('info');
+      expect(content.info).to.have.property(
+        'title', 'test-app');
+      expect(content).to.have.property('tags');
+      expect(content.consumes).to.have.members([
+        'application/json',
+        'application/x-www-form-urlencoded',
+        'application/xml', 'text/xml'
+      ]);
+      expect(content.produces).to.have.members([
+        'application/json',
+        'application/xml', 'text/xml',
+        'application/javascript', 'text/javascript'
+      ]);
+      done();
+    });
+  });
+
+  it('YAML format', function(done) {
+    var gen = givenGenerator();
+    gen.options.output = 'swagger.yaml';
+
+    gen.run(function() {
+      var content = readYamlFile();
```

Comment:
Nitpick: since YAML is a superset of JSON, this would pass even if we were producing only JSON files. Makes me wonder if there is any reasonably easy way how to check that the file actually uses YAML features that are not in JSON?